File: 2024/day1/part2.lua
dofile("UsefulStuff.lua")

local function part2(fileName)
	local list1 = {}
	local list2 = {}

	for line in io.lines(Script_path() .. fileName) do
		local parser = string.gmatch(line, "(%d+)")
		table.insert(list1, parser())
		local num = parser()
		list2[num] = (list2[num] or 0) + 1
	end

	local total = 0
	for _,v in ipairs(list1) do
		total = total + v * (list2[v] or 0)
	end
	return total
end

assert(part2("testcases/test1.txt") == 31, "Test 1 failed")
local result = part2("input.txt")
assert(result == 18805872, "Wrong answer")
print("Part 2: " .. tostring(result))

File: 2025/day1/Javascript/Part2.js
const { assert } = require('console');
const fs = require('fs');

function part2(filepath) {
    zeros = 0
    dial = 50
    fs.readFileSync(filepath, 'UTF8').toString().split('\n').forEach(element => {
        op =  new Object({
            dir: element.charAt(0),
            num: element.substring(1)
        })

        while(op.num > 0){
            dial = (100 + (op.dir == "L" ? dial-1 : dial+1)) % 100
            op.num--
            if(dial == 0)
                zeros++
        }
    });
    return zeros
}

assert(part2(__dirname+'/../testcases/test1.txt')==6)
assert(part2(__dirname+'/../input.txt')==6228)
console.log("Part 2:",part2(__dirname+'/../input.txt'))

